docs/how-it-works.md

```diff
@@ -0,0 +1,59 @@
+
+One of the uses of the model is avoid unnecessary reloads when there's no change in the state and to detect conflicts in definitions.
+
+The final representation of the model is generated from a [Go template](6) and the model that reflects the NGINX configuration.
+
+## Building the NGINX model
+
+Building a model is an expensive operation, for this reason, the use of the synchronization loop is a must. By using a [work queue](4) it is possible to not lose changes and remove the use of [sync.Mutex](5) to force a single execution of the sync loop and additionally it is possible to create a time window between the start and end of the sync loop that allows us to discard unnecessary updates. It is important to understand that any change in the cluster could generate events that the informer will send to the controller and one of the reasons for the [work queue](4).
+
+Operations to build the model:
+
+- List Ingress rules using the field ResourceVersion, i.e. old rules first.
+  - If the same path for the same host is defined in more than one Ingress, the oldest rule wins.
+  - If more than one Ingress defines a TLS section for the same host, the oldest rule wins.
+  - If multiple Ingresses define an annotation that affects configuration of the Server block, the oldest rule wins.
+
+- Create a list of NGINX Servers (hostname)
+- Create a list of NGINX Upstreams
+- If multiple Ingresses define different paths for the same host, the ingress controller will merge the definitions.
+- Annotations are applied to all the paths in the Ingress.
+- Multiple Ingresses can define different annotations. These definitions are not shared between Ingresses.
+
+## When a reload is required
+
+The next list describe the scenarios when a reload is required:
+
+- New Ingress Resource Created.
+- TLS section is added to existing Ingress.
+- Change in Ingress annotations.
+- A path is added/removed from an Ingress.
+- An Ingress, Service, Secret is removed.
+- Some missing referenced object from the Ingress is available, like a Service, Secret or Endpoint.
+- A Secret is updated.
+
+## Avoiding reloads
+
+In some cases, it is possible to avoid reloads, in particular when there is a change in the endpoints, i.e. a pod is started or replaced. It is out of the scope of this Ingress controller to remove reloads completely. This would require an incredible amount of work and at some point makes no sense. This can change only if NGINX changes the way new configurations are read, basically, new changes do not replace worker processes.
```

Comment:
should we reference to enable-dynamic-confioguration docs here?

Reply:
Not sure, maybe when we enable enable-dynamic-configuration as the default?

Reply:
OK, 

> In some cases, it is possible to avoid reloads, in particular when there is a change in the endpoints, i.e. a pod is started or replaced

is a bit confusing then. It sounds like we are saying ingress-nginx already avoids reload in those cases by default - but it does not.